library/lua/gui/widgets.lua

```diff
@@ -1009,23 +1015,22 @@ function EditField:onInput(keys)
     elseif keys.KEYBOARD_CURSOR_LEFT then
         self:setCursor(self.cursor - 1)
         return true
-    elseif keys.CUSTOM_CTRL_B then -- back one word
+    elseif keys.CUSTOM_CTRL_LEFT then -- back one word
         local _, prev_word_end = self.text:sub(1, self.cursor-1):
                                                find('.*[%w_%-][^%w_%-]')
         self:setCursor(prev_word_end or 1)
         return true
-    -- commented out until we get HOME key support from DF
-    -- elseif keys.CUSTOM_CTRL_A then -- home
-    --     self:setCursor(1)
-    --     return true
+    elseif keys.CUSTOM_HOME then
+        self:setCursor(1)
+        return true
     elseif keys.KEYBOARD_CURSOR_RIGHT then
         self:setCursor(self.cursor + 1)
         return true
-    elseif keys.CUSTOM_CTRL_F then -- forward one word
+    elseif keys.CUSTOM_CTRL_RIGHT then -- forward one word
         local _,next_word_start = self.text:find('[^%w_%-][%w_%-]', self.cursor)
         self:setCursor(next_word_start)
         return true
-    elseif keys.CUSTOM_CTRL_E then -- end
+    elseif keys.CUSTOM_END then
```

Comment:
The reason I replaced the emacs bindings was precisely because they were restricting use of commonly used shortcuts. When an edit field has permanent focus (common for tools with search), all other shortcuts become Ctrl-something. The emacs cursor navigation shortcuts are taking up prime real estate. 

Ctrl-A for beginning of line was already disabled because several tools needed that for "select all".

I'm not sure what the best path forward is.